### PR TITLE
tiltfile2: make it more clear when k8s_yaml accepts a string

### PR DESCRIPTION
**Pull request by jazzdan**

### CODE REVIEW

**Comment by nicks** [comment]
I'm confused...the PR description is "k8s yaml accepts string", but the PR is "k8s yaml does not accept string"?

**Comment by jazzdan** [comment]
Whoops, sorry @nicks that was an old title. Re-titling.

File: internal/tiltfile2/tiltfile_test.go [hunk 1]
```diff
@@ -746,6 +746,20 @@ services:
...
 }
 
+func TestK8sYAMLInput(t *testing.T) {
+	f := newFixture(t)
+	defer f.TearDown()
+
+	f.setupFoo()
+	f.WriteFile("bar.yaml", "im not yaml")
+	f.file("Tiltfile", `
+k8s_yaml('bar.yaml')
```
Comment (nicks) [inline internal/tiltfile2/tiltfile_test.go]: can you also add a test case for something like

```
k8s_yaml(str(read_file('foo.yaml')))
```

File: internal/tiltfile2/k8s.go [hunk 2]
```diff
@@ -132,10 +133,22 @@ func (s *tiltfileState) yamlEntitiesFromSkylarkValue(v skylark.Value) ([]k8s.K8s
 		if err != nil {
 			return nil, err
 		}
-		return k8s.ParseYAMLFromString(string(bs))
+		entities, err := k8s.ParseYAMLFromString(string(bs))
```
Comment (nicks) [inline internal/tiltfile2/k8s.go]: would be slightly more idiomatic for the err != nil check to live here instead of the handle function

**Pull request merged**